File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        majority = -1
        for num in nums:
            if num == majority:
                count += 1
                continue
            if count == 0:
                majority = num
                count = 1
            count -= 1
        return majority
    # def majorityElement(self, nums: List[int]) -> int:
    #     numbers = Counter(nums)
    #     for key,value in numbers.items():
    #         if value > len(nums) // 2:
    #             return key
    #     return result

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int majority = 0;
        int number = nums[0];
        for(int num : nums){
            if(majority == 0){
                number = num;
                majority += 1;
            }else if(number == num){
                majority += 1;
            }else{
                majority -= 1;
            }
        }
        return number;        
    }
}